src/scripts/main.js

```diff
@@ -1,3 +1,335 @@
+        <option value="San Francisco">San Francisco</option>
+      </select>
+    </label>
+    <label>Age:
+      <input name="age" data-qa="age" type="number" required>
+    </label>
+    <label>Salary:
+      <input name="salary" data-qa="salary" type="number" required>
+    </label>
+    <button type="submit">Save to table</button>
+  </form>
+`);
+
+const form = document.forms['employee-form'];
+const lastElem = form.elements[form.elements.length - 1];
+const firstElem = form.elements[0];
+
+lastElem.onkeydown = function(e) {
+  if (e.key === 'Tab' && !e.shiftKey) {
+    firstElem.focus();
+
+    return false;
+  }
+};
+
+firstElem.onkeydown = function(e) {
+  if (e.key === 'Tab' && e.shiftKey) {
+    lastElem.focus();
+
+    return false;
+  }
+};
+
+lastElem.addEventListener('click', e => {
+  e.preventDefault();
+
+  const data = Object.fromEntries(new FormData(form).entries());
+
+  if (!validationsOfInformation(data)) {
+    return;
+  }
+
+  saveDataToTable(data);
+  removeSelect();
+  form.reset();
+});
+
+function saveDataToTable(data) {
+  const tr = document.createElement('tr');
+
+  tr.innerHTML = `
+    <td>${data.name}</td>
+    <td>${data.position}</td>
+    <td>${data.office}</td>
+    <td>${data.age}</td>
+    <td>$${Number(data.salary).toLocaleString('en-US')}</td>
+  `;
+
+  table.append(tr);
+}
+
+const minAge = 18;
+const maxAge = 90;
+
+function validationsOfInformation(data) {
+  if (data.name.length < 4) {
+    pushNotification(
+      'Error',
+      'Name must contain more then 3 character',
+      'error'
+    );
+
+    return false;
+  }
+
+  if (data.age < minAge || data.age > maxAge) {
+    pushNotification(
+      'Error',
+      'Age must be between 18 and 90',
+      'error'
+    );
+
+    return false;
+  }
+
+  if (data.position === '' || data.salary === '') {
+    pushNotification(
+      'Error',
+      'Fill in all fields of the form',
+      'error'
+    );
+
+    return false;
+  }
+
+  pushNotification('Success',
+    'New employee successfully added',
+    'success'
+  );
+
+  return true;
+}
+
+function pushNotification(title, description, type) {
+  document.body.insertAdjacentHTML('beforeend', `
+    <div class="notification ${type}" data-qa="notification">
+      <h2 class="title">${title}</h2>
+      <p>${description}</p>
+  </div>
+`);
+
+  setTimeout(() => {
+    document.body.querySelector('.notification').remove();
+  }, 2000);
+};
+
+let editingTd;
+
+table.addEventListener('dblclick', e => {
+  e.preventDefault();
+
+  if (e.target.tagName !== 'TD') {
+    return;
+  }
+
+  const td = e.target;
+  const index = td.cellIndex;
+
+  makeTdEditable(td, index);
+});
+
+function makeTdEditable(td, index) {
+  editingTd = {
+    elem: td,
+    data: td.innerHTML,
+  };
+
+  let input;
+
+  switch (index) {
+    case 0:
+      input = document.createElement('input');
+      input.type = 'text';
+      input.placeholder = 'at least 4 character';
```

Comment:
1) move `4` to a constant
```
const MIN_CHARACTERS = 4
```
2) use for template string
```suggestion
      input.placeholder = `at least ${MIN_CHARACTERS} characters`;
```